### ISSUE TITLE
Unexpected token . error

### ISSUE THREAD

**Issue by omerts** [eslint/espree#214]
Hello,

I am getting: 
```
Unexpected token .
```

For the following, working code:
```javascript
let _legend = this.props.items.map((item, i) => {
       var itemStyle = { 'background': this.props.colors(i) };

      return(
          <div className="legend-item" key={i} data-horizontal data-layout>
            <div className="legend-item-color" style={itemStyle}/>
            <h4 data-flex>{this._capitalizeFirstLetter(item.name)}</h4>
            <h4>{item.value}</h4>
          </div>
        );
    }.bind(this));
```
The problem espree has with the above code, it that the arrow function is not surrounded by parenthesis, even though the above code works fine. 

This code has no error: 
```javascript
let _legend = this.props.items.map(((item, i) => {
       var itemStyle = { 'background': this.props.colors(i) };

      return(
          <div className="legend-item" key={i} data-horizontal data-layout>
            <div className="legend-item-color" style={itemStyle}/>
            <h4 data-flex>{this._capitalizeFirstLetter(item.name)}</h4>
            <h4>{item.value}</h4>
          </div>
        );
    }).bind(this));
```

But the first snippet of code is valid, so imo espree should be able to parse it fine.

**Comment by nzakas** [comment]
As far as I know, Espree is behaving correctly here. You cannot have `.bind` without surrounding the arrow function with parens. If it's "working" for you without parens, then it's likely a bug in whatever parser/engine you're using.

**Comment by omerts** [comment]
I am using babel

**Comment by nzakas** [comment]
Then you should file a bug with Babel.
